### PR TITLE
[UnitTests] Added unit tests for output tracking.

### PR DESCRIPTION
**Pull request by mkrueger**
Added unit tests for output tracking

### CODE REVIEW

**Comment by slluis** [comment]
This PR is incorrectly bumping debugger-libs

File: main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			public TrackTestProject (string language, string type) : base(language)
+			{
+				this.type = type;
+				Initialize (this);
+			}
+
+		}
+
+		[Test]
+		public void TestOuptutTracking_ProjectType ()
```
Comment (KirillOsenkov) [inline main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs]: Ouptut is misspelled

File: main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			TypeSystemService.AddOutputTrackingNode (new TypeSystemOutputTrackingNode { ProjectType = "TestProjectType" });
+
+			Assert.IsFalse (TypeSystemService.IsOutputTrackedProject (new TrackTestProject ("C#", "Bar")));
+			Assert.IsTrue (TypeSystemService.IsOutputTrackedProject (new TrackTestProject ("C#", "TestProjectType")));
+		}
+
+		[Test]
+		public void TestOuptutTracking_LanguageName ()
+		{
+			TypeSystemService.AddOutputTrackingNode (new TypeSystemOutputTrackingNode { LanguageName = "IL" });
```
Comment (KirillOsenkov) [inline main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs]: Are these tests mutating static state? Should they restore the state to what it was previously? If these tests run in parallel or multiple times, will it still work? Consider adding some sort of internal Reset method to TypeSystemService?

File: main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		public void TestOuptutTracking_ProjectType ()
+		{
+			TypeSystemService.AddOutputTrackingNode (new TypeSystemOutputTrackingNode { ProjectType = "TestProjectType" });
+
+			Assert.IsFalse (TypeSystemService.IsOutputTrackedProject (new TrackTestProject ("C#", "Bar")));
+			Assert.IsTrue (TypeSystemService.IsOutputTrackedProject (new TrackTestProject ("C#", "TestProjectType")));
+		}
+
+		[Test]
+		public void TestOuptutTracking_LanguageName ()
```
Comment (KirillOsenkov) [inline main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs]: Ouptut is misspelled :)

File: main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs (hunk 2 above)
Comment (mkrueger) [inline main/tests/UnitTests/MonoDevelop.Ide/TypeSystemServiceTests.cs]: Doesn't really matter here.

**Pull request merged**